### ISSUE TITLE
Detect post language

### ISSUE THREAD

**Issue by kaiserkiwi** [feditext/feditext#170]
**Is your feature request related to a problem? Please describe.**
As I type the composer should detect the language I'm typing in.

**Describe the solution you'd like**
When German is selected in the composer but I write in English the language selection should be highlighted to inform me that the current language and the written language differ from each other.

**Describe alternatives you've considered**
An automated switch of the post language would (optionally?) an alternative.

**Additional context**
Apps like IceCubes or the official Mastodon apps are able to do this.

**Comment by VyrCossont** [comment]
@kaiserkiwi Thanks for the feedback here and on the other language-related posts. For reference, when you write in English or German, do you select a different software keyboard type? And on a related note, would you expect the app to change the software keyboard type for you if you changed the post language?

**Comment by VyrCossont** [comment]
We've got at least two options here, [`NLLanguageRecognizer`](https://developer.apple.com/documentation/naturallanguage/nllanguagerecognizer) and [`UITextInputMode.primaryLanguage`](https://developer.apple.com/documentation/uikit/uitextinputmode/1614535-primarylanguage), and the latter is going to be much easier on battery life than the former, but it won't work with hardware keyboards.

[`UIResponder.textInputMode`](https://developer.apple.com/documentation/uikit/uiresponder/1621133-textinputmode) could be used to switch the software keyboard language.

**Comment by kaiserkiwi** [comment]
To be more clear. I mean this selection. I usually don't switch the keyboard layout when I write in English. 

![image](https://github.com/feditext/feditext/assets/8428551/0c9b2982-5c39-41d1-a8f6-aca548c418e0)

The reason here is accessibility. If I reply to an English post and my post language is German and I don't realize it, than everyone with a screen reader will have a pretty bad time.

**Comment by VyrCossont** [comment]
I'm just speculating about how to do it efficiently. The official iOS app is definitely using `NLLanguageRecognizer`, the relatively expensive approach.

**Comment by kaiserkiwi** [comment]
I don't think you have to call it on ever keystroke. Count words, when more than 5 check the language. 

I don't know if the API returns a probability score but if it's too low you could check after 5 additional words again.

Or just in general: Check every five words. IceCubes is open source too. (And the official app too, right?) Maybe you can get inspiration there.
